dss/operations/checkout.py

```diff
@@ -0,0 +1,188 @@
+                for key in self.handle.list(self.bucket, _key):  # handles checkout/bundle/*
+                    self._verify_delete(self.handle, self.bucket, key)
+                uuid, version = self._parse_key(_key)
+                manifest = get_bundle_manifest(replica=self.replica, uuid=uuid, version=version)
+                if manifest is None:
+                    logger.warning(f"unable to locate manifest for fqid: {self.bucket}/{_key}")
+                    continue
+                for _files in manifest['files']:
+                    key = compose_blob_key(_files)
+                    self._verify_delete(self.handle, self.bucket, key)
+            elif FILE_PREFIX in _key:
+                # should handle other keys, files/blobs
+                file_metadata = self._verify_get(self.handle, self.replica.bucket, _key)
+                self._verify_delete(self.handle, self.bucket, key=compose_blob_key(file_metadata))
+
+
+@checkout.action("verify",
+                 arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
+                            "--keys": dict(nargs="+", help="keys to check the status of", required=True)})
+class verify(CheckoutHandler):
+    def process_keys(self):
+        """
+        Verify that keys are in the checkout bucket
+        """
+        checkout_status = dict(replica=self.replica.name)
+        for _key in self.keys:
+            if DSS_BUNDLE_KEY_REGEX.match(_key):  # handles bundles/fqid keys or fqid
+                uuid, version = self._parse_key(_key)
+                bundle_manifest = get_bundle_manifest(replica=self.replica, uuid=uuid, version=version)
+                checkout_bundle_contents = [x[0] for x in self.handle.list_v2(bucket=self.bucket,
+                                                                              prefix=f'bundles/{uuid}.{version}')]
+                bundle_internal_status = list()
+
+                for _file in bundle_manifest['files']:
+                    temp = collections.defaultdict(blob_checkout=False, bundle_checkout=False, should_be_cached=False)
+                    bundle_key = f'bundles/{uuid}.{version}/{_file["name"]}'
+                    blob_key = compose_blob_key(_file)
+
+                    blob_status = self._verify_blob_existance(self.handle, self.bucket, blob_key)
+                    if blob_status:
+                        temp['blob_checkout'] = True
+                    if bundle_key in checkout_bundle_contents:
+                        temp['bundle_checkout'] = True
+                    if cache_flow.should_cache_file(_file['content-type'], _file['size']):
+                        temp['should_be_cached'] = True
+
+                    for x in ['name', 'uuid', 'version']:
+                        temp.update({x: _file[x]})
+                    bundle_internal_status.append(temp)
+                checkout_status[_key] = bundle_internal_status
+            elif FILE_PREFIX in _key:
```

Comment:
Should be elif _key.startswith(FILE_PREFIX). Odds are low that "files" might be a part of the uuid, but it's not impossible.